src/types/onyx/Transaction.ts

```diff
@@ -40,80 +41,174 @@ type Waypoint = {
     originalTransactionID?: string;
+
+    /** In split transactions this is a collection of participant split data */
     splits?: Split[];
 };
 
+/** Model of transaction custom unit */
 type TransactionCustomUnit = {
+    /** ID of the custom unit */
     customUnitID?: string;
+
+    /** ID of the custom unit rate */
     customUnitRateID?: string;
+
+    /** Custom unit amount */
     quantity?: number;
+
+    /** TODO: I think this value can be changed to `ValueOf<CONST.CUSTOM_UNITS>` */
+    /** Name of the custom unit */
     name?: string;
+
+    /** Default rate for custom unit */
     defaultP2PRate?: number;
 };
 
+/** Types of geometry */
 type GeometryType = 'LineString';
 
+/** Geometry data */
 type Geometry = {
+    /** Matrix of points, indexed by their coordinates */
     coordinates: number[][] | null;
+
+    /** Type of connections between coordinates */
     type?: GeometryType;
 };
 
+/** Accepted receipt paths */
 type ReceiptSource = string;
 
+/** Model of receipt */
 type Receipt = {
+    /** TODO: This doesn't exist in the app */
     receiptID?: number;
+
+    /** TODO: This doesn't exist in the app */
     path?: string;
+
+    /** TODO: This doesn't exist in the app */
     name?: string;
+
+    /** Path of the receipt file */
     source?: ReceiptSource;
+
+    /** Name of receipt file */
     filename?: string;
+
+    /** Current receipt scan state */
     state?: ValueOf<typeof CONST.IOU.RECEIPT_STATE>;
+
+    /** Type of the receipt file */
     type?: string;
 };
 
+/** Model of route */
 type Route = {
+    /** Distance amount of the route */
     distance: number | null;
+
+    /** Route geometry data */
     geometry: Geometry;
 };
 
+/** Collection of routes, indexed by `route${index}` where `index` is the position of the route in the list */
 type Routes = Record<string, Route>;
 
-type ReceiptError = {error?: string; source: string; filename: string};
+/** Model of receipt error */
+type ReceiptError = {
+    /** TODO: This doesn't exist in the app */
+    error?: string;
 
+    /** Path of the receipt file */
+    source: string;
+
+    /** Name of the receipt file */
+    filename: string;
+};
+
+/** Collection of receipt errors, indexed by a UNIX timestamp of when the error occurred */
 type ReceiptErrors = Record<string, ReceiptError>;
 
+/** Tax rate data */
 type TaxRateData = {
+    /** TODO: This doesn't exist in the app */
     name: string;
+
+    /** Tax rate percentage */
     value: string;
+
+    /** Tax rate code */
     code?: string;
 };
 
+/** Model of tax rate */
 type TaxRate = {
+    /** Default name of the tax rate */
     text: string;
+
+    /** Key of the tax rate to index it on options list */
     keyForList: string;
+
+    /** TODO: This doesn't exist in the app */
     searchText: string;
+
+    /** TODO: This doesn't exist in the app */
     tooltipText: string;
+
+    /** TODO: This doesn't exist in the app */
     isDisabled?: boolean;
```

Comment:
A rule of thumb for properties that doesn't exist is to check what are the real values for these, if they exist I think it's better to add a comment explaining these properties